lib/spack/docs/packaging_guide.rst

```diff
@@ -1976,8 +1977,10 @@ The following dependency types are available:
   (building project X doesn't need project Y's build dependencies).
 * **"link"**: the project is linked to by the project. The package will be
   added to the current package's ``rpath``.
-* **"run"**: the project is used by the project at runtime. The package will
-  be added to ``PATH`` and ``PYTHONPATH``.
+* **"run"**: the listed project is used by the target project at runtime.
+  The package will be added to ``PATH`` and ``PYTHONPATH``.
+* **"test"**: the listed project is used by the target project's testing
+  system.
```

Comment:
Wow, this whole section of docs is terrible. "the project is used by the project" is so confusing. Let me tackle this in a separate PR. This also relates to #3768.

Reply:
@adamjstewart Should I remove this file from my PR?

Reply:
Yes please, I'll ping you on the new PR.